lightbulb/command_handler.py

```diff
@@ -169,6 +175,27 @@ def __init__(
 
         self._help_impl = help_class(self)
 
+        self.extensions_path = extensions_path
+        if self.extensions_path:
+            self.full_extensions_path = pathlib.Path.cwd() / self.extensions_path.replace(".", os.sep)
+
+            self.subscribe(hikari.StartedEvent, self.start_extension_watch)
+
+    async def start_extension_watch(self, _) -> None:
+        async for changes in awatch(self.full_extensions_path):
+            for change in changes:
+                change_type = change[0]
+                change_path = change[1]
+
+                file_changed = (change_path.split(os.sep)[-1]).split(".")[0]
+                extension_path = self.extensions_path + "." + file_changed
+
+                if change_type == Change.deleted:
+                    self.unload_extension(extension_path)
+
+                if change_type == Change.modified:
+                    self.reload_extension(extension_path)
```

Comment:
Should you also load new extensions?

Reply:
When you create a file you won't have a load function yet

Reply:
@YodaPY why not? Also, just realized that there is no error handling. You should probably handle errors when loading/reloading/unloading and just log it

Reply:
When you create a file it counts as `Change.ADDED`, anything that happens after that e.g. writing the actual code and then saving counts as Change.MODIFIED. Also IMO the error handling and logging should happen in the load-extension methods, not here.

Reply:
No, those methods should raise the errors not handle them, which is currently what happens

Reply:
Sorry understood it wrong. Yeah I can do that.

Reply:
> No, those methods should raise the errors not handle them, which is currently what happens

I haven't tested the code, but wouldn't that crash the whole bot and require you to re-run the script?

Reply:
yes, and thats intended unless you catch them yourself